packages/libs/deer/README.md

```diff
@@ -0,0 +1,162 @@
+
+`deer` solves this problem, by returning every issue present.
+This means that a single API call with the payload given will result in the
+errors: `257 larger than i8::MAX`, `null is not String`, and `extra key "extra" provided`.
+
+This in turn also means that deer can be used for custom validation while
+deserializing, while still being able to return all validation issues.
+
+<sub>
+deer might provide a way in the future to describe these constraints.
+</sub>
+
+## Limitations
+
+deer currently does **not** parse values itself, but relies on external parsers
+like `serde_json`, this means that parsing will be fail-fast, and deer only touches
+syntactically correct values.
+
+## Future Plans
+
+The first release of `deer` is intentionally minimal and tries to lay a good foundation to
+extend functionality in the future.
+There are many future possible directions and ideas we're trying to see if they can
+benefit the different use cases.
+
+### Introspection Support
+
+Currently, popular crates like `serde` do not provide a way to introspect what the output
+will be. Other tools try to fill the gap by manually interpreting the instructions given
+to these crates. This often leads to edge cases, resulting in the dissonance between the
+expected value and reality. The goal with the explicit support of introspection is to
+allow other tools to make use of it and build abstractions around it instead of trying to
+reverse engineer.
+
+<sup>
+How the introspected format might look like is currently unknown.
+</sup>
+
+### Validation
+
+Currently, to be able to do validation, one must create a new type that performs the
+validation step. This extra boilerplate is often a pain point. The idea is to instead
+allow for _optional_ validation via combinators using
+the derive macro.
+
+<details>
+<summary>What it may look like</summary>
+
+```rust
+#[derive(deer::Deserialize)]
+struct Payload {
+    #[validate(min(12) & max(24))]
+    length: u8
+}
+```
+
+</details>
+
+### Lax Deserialization
+
+Instead of strictly deserializing types, one might prefer to deserialize while coercing
+values (`"1"` might be interpreted as `1` instead),
+This behavior would be opt-in instead of opt-out and enabled on the `derive` level.
+
+### Deserializer with Parser
+
+`deer` currently relies on external tools (notably `serde`) to implement parsing of
+different formats, which means that we still fail-fast during the parsing of malformed
+input.
+In the future, `deer` might provide its own parser, which tries to recover from parsing
+errors and still provide parsing diagnostics.
+
+<details>
+<summary>What it may look like</summary>
+
+```json
+{
+  "i8": "string"
+```
+
+`deer` currently relies on external tools (notably `serde`) to implement parsing of
+different formats, which means that we still fail fast during the parsing of malformed
+input.
+In the future, `deer` might provide a parser that tries to recover from parse errors and
+provide meaningful diagnostics.
```

Comment:
This section is almost a duplicate of the above

Reply:
kind of. I want to convey that the current limitation of failing fast on parsing might be lifted in the future. Maybe rewording is needed.